### PR TITLE
Feature/18299 pupils not taking check data service

### PR DESCRIPTION
**Pull request by GuyHarwood**
very small PR.  most of the work is done in the pupil data service refactor.

### CODE REVIEW

File: admin/controllers/school.js [hunk 1]
```diff
@@ -332,14 +332,14 @@ const getPupilNotTakingCheck = async (req, res, next) => {
 
   // Get attendance code index
   try {
-    attendanceCodes = await pupilsNotTakingCheckDataService.getAttendanceCodes()
+    attendanceCodes = await pupilsNotTakingCheckDataService.sqlGetAttendanceCodes()
```
Comment (jon-shipley) [inline admin/controllers/school.js]: method name: should `sqlFindAttendanceCodes`

File: admin/controllers/school.js [hunk 2]
```diff
@@ -332,14 +332,14 @@ const getPupilNotTakingCheck = async (req, res, next) => {
...
   } catch (error) {
     return next(error)
   }
 
   // Get pupils for active school
   try {
-    pupils = await pupilsNotTakingCheckDataService.fetchPupilsWithReasons(req.user.School)
+    pupils = await pupilsNotTakingCheckDataService.sqlFetchPupilsWithReasons(req.user.School)
```
Comment (jon-shipley) [inline admin/controllers/school.js]: method name: should `sqlFindPupilsWithReasons`

File: admin/controllers/school.js [hunk 3]
```diff
@@ -382,7 +382,7 @@ const getSelectPupilNotTakingCheck = async (req, res, next) => {
 
   // Get attendance code index
   try {
-    attendanceCodes = await pupilsNotTakingCheckDataService.getAttendanceCodes()
+    attendanceCodes = await pupilsNotTakingCheckDataService.sqlGetAttendanceCodes()
```
Comment (jon-shipley) [inline admin/controllers/school.js]: method name: should `sqlFindAttendanceCodes`

File: admin/controllers/school.js [hunk 4]
```diff
@@ -460,14 +460,14 @@ const savePupilNotTakingCheck = async (req, res, next) => {
...
```
Comment (jon-shipley) [inline admin/controllers/school.js]: method name: should `sqlFindAttendanceCodes`

File: admin/controllers/school.js [hunk 5]
```diff
@@ -460,14 +460,14 @@ const savePupilNotTakingCheck = async (req, res, next) => {
...
   } catch (error) {
     return next(error)
   }
 
   // Get pupils for active school
   try {
-    pupils = await pupilsNotTakingCheckDataService.fetchPupilsWithReasons(req.user.School)
+    pupils = await pupilsNotTakingCheckDataService.sqlFetchPupilsWithReasons(req.user.School)
```
Comment (jon-shipley) [inline admin/controllers/school.js]: method name: should sqlFindPupilsWithReasons

File: admin/services/data-access/pupils-not-taking-check.data.service.js [hunk 6]
```diff
@@ -2,27 +2,55 @@
...
+ * @returns {Promise.<*>}
+ */
+  getAttendanceCodes: async () => {
+    return AttendanceCode
+      .find()
+      .sort('order')
+      .lean()
+      .exec()
+  },
+  sqlFindAttendanceCodes: async () => {
```
Comment (jon-shipley) [inline admin/services/data-access/pupils-not-taking-check.data.service.js]: This seems to be in the wrong service.  Shouldn't it be in the `attendanceCodeDataService`?  Leaving it in the `pupils-not-taking-check.data.service` means that dependencies that are not needed would be pulled in by the calling module.

Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupils-not-taking-check.data.service.js]: AttendanceCodeDataService currently doesn't exist right?
Extracting a new service which indeed does make sense would mean doing more than what the ticket is about.
It does seem a small change however so why not.

Comment (GuyHarwood) [reply admin/services/data-access/pupils-not-taking-check.data.service.js]: @jon-shipley @GeorgeChatzigiannis yes its an odd set of functions in the one data service.  plus, the other attendance status functions are scattered around, some of which i believe are in other pupil related data services. messy

Comment (GuyHarwood) [reply admin/services/data-access/pupils-not-taking-check.data.service.js]: `attendanceCodeDataService` created

File: admin/controllers/school.js [hunk 7]
```diff
@@ -332,14 +333,14 @@ const getPupilNotTakingCheck = async (req, res, next) => {
 
   // Get attendance code index
   try {
-    attendanceCodes = await pupilsNotTakingCheckDataService.getAttendanceCodes()
+    attendanceCodes = await attendanceCodeDataService.sqlFindAttendanceCodes()
```
Comment (jon-shipley) [inline admin/controllers/school.js]: 👍 better

**Pull request merged**